`Development`: Add branch naming conventions to documentation (#8504)
ls1intum/Artemis: docs/dev/development-process/development-process.rst

For more details and to get started with UICompanion, visit the `UICompanion Repository on GitHub <https://github.com/ls1intum/UICompanion>`_.

4. Create a local Branch
========================
We use the following structure for branch names: ``<type>/<area>/<short-description>``

> - Possible types are: **feature**, **chore** and **bugfix**

Reviewer: ```suggestion
- Possible types are: **feature**, **chore**, and **bugfix**
```

- Possible areas are all allowed feature tags (see :ref:`pr_naming_conventions`), written kebab-case. For example, ``Programming exercises`` becomes **programming-exercises**.

> **Branches that do not follow this structure will get rejected by GitHub!**

Reviewer: ```suggestion
**Branches that do not follow this structure will get rejected automatically!**
```

For more details and to get started with UICompanion, visit the `UICompanion Repository on GitHub <https://github.com/ls1intum/UICompanion>`_.

4. Create a local Branch
========================
We use the following structure for branch names: ``<type>/<area>/<short-description>``
- Possible types are: **feature**, **chore** and **bugfix**
- Possible areas are all allowed feature tags (see :ref:`pr_naming_conventions`), written kebab-case. For example, ``Programming exercises`` becomes **programming-exercises**.

**Branches that do not follow this structure will get rejected by GitHub!**

5. Implement the Feature
========================
In this step, the development team converts the detailed plans and designs outlined in the functional proposal into working code.
This step requires careful adherence to the previously defined requirements and system architecture to ensure that the function fits seamlessly into the existing system and fulfills the specified functional and performance criteria.

> Naming Conventions for Branches
> --------------------------------
> 

Author: ```suggestion
```
